solo-io/gloo
# Feat/lambda with basic defaults

# Description

New Lambda functionality to enable route level overrides as well as a way to use the first function for uses cases such as older portal that cant set all the fields for aws specs on routes.

# Context

This PR does not attempt to change anything about envoy configuration for those who have not opted in or set new options.
The ability to use the first function should only occur if and when helm configuration is set as it is a behavior change that we likely don't want in general. 
https://github.com/solo-io/solo-projects/pull/4313 implements the breaking function signature change in ee.

# Testing Notes
The first consideration in this PR is routes without destinationspec for functions. This should only occur if the helm setting is enabled.
To test take a valid setup with 1 function. Go to the route and delete the function specifier. Expect an error. Update settings to allow for defaults expect to get back to the same envoy configuration.

The second consideration is defaults. Take a valid setup with a destinationspec that has at least 1 non-default value, for example request transformation. 
1. Set that non-default value on the upstream's defaults. Nothing should change. 
2. Set that non-default value back to default. Nothing should change
3. Set the non-default value back to default on the upstream defaults. The configuration should change.
